strax/plugins/down_chunking_plugin.py

```diff
@@ -0,0 +1,40 @@
+import strax
+import types
+import inspect
+from .plugin import Plugin
+export, __all__ = strax.exporter()
+
+
+##
+# Plugin which allows to use yield in plugins compute method.
+# Allows to chunk down output before storing to disk.
+# Only works if multiprocessing is omitted.
+##
+
+@export
+class DownChunkingPlugin(Plugin):
+    """Plugin that merges data from its dependencies
+    """
+
+    def iter(self, iters, executor=None):
+
+        _plugin_uses_multi_threading = (self.parallel 
+                                        and executor is not None 
+                                        and (inspect.isgeneratorfunction(self.compute))
+                                       )
+        if _plugin_uses_multi_threading:
+            raise NotImplementedError(
+                f'Plugin "{self.__class__.__name__}" uses an iterator as compute method. ' 
+                'This is not supported in multi-threading/processing.')
+        return super().iter(iters, executor=None)
+
+    def _iter_return(self, chunk_i, **inputs_merged):
+        return self.do_compute(chunk_i=chunk_i, **inputs_merged)
+
+    def _fix_output(self, result, start, end, _dtype=None):
```

Comment:
Am I right that `_fix_output` is currently skipped by these plugins, and thus multi-output is not supported? That could be another check at the beginning (`assert not self.multi_output` or maybe a nicer `NotImplementedError`). You could also make `_fix_output` equal to `pass` -- or is there a reason you did it like this?